text/0232-maturity-of-otel.md

```diff
@@ -0,0 +1,72 @@
+
+Components should not be marked as stable if their user-visible interfaces are not stable. For instance, if the Collector's component "otlpreceiver" declares a dependency on the OpenTelemetry Collector API "config" package which is marked with a maturity level of "beta", the "otlpreceiver" should be at most "beta". Maintainers are free to deviate from this recommendation if they believe users are not going to be affected by future changes.
+
+For the purposes of this document, a breaking change is defined as a change that may require consumers of our components to adapt themselves in order to avoid disruption to their usage of our components.
+
+#### Development
+
+Not all pieces of the component are in place yet, and it might not be available for users yet. Bugs and performance issues are expected to be reported. User feedback is desired, especially regarding the user experience (configuration options, component observability, technical implementation details, and so on). Configuration options might break often depending on how things evolve. The component SHOULD NOT be used in production. The component MAY be removed without prior notice.
+
+#### Alpha
+
+This is the default level: any components with no explicit maturity level should be assumed to be "Alpha". The component is ready to be used for limited non-critical production workloads, and the authors of this component welcome user feedback. Bugs and performance problems are encouraged to be reported, but component owners might not work on them immediately. The component's interface and configuration options might often change without backward compatibility guarantees. Components at this stage might be dropped at any time without notice.
+
+#### Beta
+
+Same as Alpha, but the interfaces (API, configuration, generated telemetry) are treated as stable whenever possible. While there might be breaking changes between releases, component owners should try to minimize them. A component at this stage is expected to have had exposure to non-critical production workloads already during its Alpha phase, making it suitable for broader usage.
+
+#### Release Candidate
+
+The component is feature-complete and ready for broader usage. The component is ready to be declared stable, it might just need to be tested in more production environments before that can happen. Bugs and performance problems are expected to be reported, and there's an expectation that the component owners will work on them. Breaking changes, including configuration options and the component's output, are not expected to happen without prior notice unless under special circumstances.
+
+#### Stable
+
+The component is ready for general availability. Bugs and performance problems should be reported, and there's an expectation that the component owners will work on them. Breaking changes, including configuration options and the component's output, are not expected to happen without prior notice unless under special circumstances.
```

Comment:
I think this question is not directly related to the definition of stability, but it is good opportunity to discuss how we understand _stable_ component which not  support all options form specifications.

I suppose that it is broader project, but lets start with .NET example:
[OTLP exporter for .NET ](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md) is declared stable now. 

It missing some parts of required configuration options like [`OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` `OTEL_EXPORTER_OTLP_METRICS_ENDPOINT` `OTEL_EXPORTER_OTLP_LOGS_ENDPOINT`](https://github.com/open-telemetry/opentelemetry-specification/blob/89f9bf3b3141626229ac39c9437e1832b38f104f/specification/protocol/exporter.md?plain=1#L7-L17) and [couple others](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4154).

Can it be called stable based on this document? Is it part of maintainer judgement or based on definition it should be never go beyond _Development_ level ("Not all pieces of the component are in place yet")?

---
@cijothomas. FYI

Reply:
Not exactly the parts you mentioned, but we had a similar discussion here some months ago and I started an issue to tackle this specific aspect: https://github.com/open-telemetry/opentelemetry-specification/issues/3673

I'm marking this as resolved, but if you believe that discussion doesn't relate to your concern, feel free to reopen.